src/ConnectedMode/Migration/MSBuildFileCleaner.ElementAndValueTailMatcher.cs

```diff
@@ -0,0 +1,59 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Migration
+{
+    internal partial class MSBuildFileCleaner
+    {
+        private static class ElementAndValueTailMatcher
+        {
+            /// <summary>
+            /// Returns a list of nodes in the document matching the following criteria:
+            ///   <ElementName ... ></ElementName>"xxxx[value to tail match]</ElementName>
```

Comment:
This class follows a similar pattern to the other helper class. but the condition it is looking for is simpler since we just need to check the element name and the value it contains.